### PR TITLE
[msbuild] Fixed provisioning profile lookups for tvOS

### PR DESCRIPTION
**Pull request by jstedfast**

### CODE REVIEW

**Comment by spouliot** [comment]
This needs more context, e.g.
* what issue does it solve : bugzilla, GitHub, other...
* if it does not fail a test, then we're likely in need of one

**Comment by jstedfast** [comment]
It fails some tvOS msbuild unit tests

**Comment by rolfbjarne** [comment]
It fixes this MSBuild unit test on wrench:

```
1) Test Failure : Xamarin.iOS.Tasks.TVAppTests("TV","iPhone").BasicTest
     #RunTarget-ErrorCount
	Could not locate the provisioning profile with a Name or UUID of e822ee59-a61b-484c-a708-f0de7c938613.
  Expected: 0
  But was:  1
```

It claims to fail a provisioning profile it found earlier:

```
Task "DetectSigningIdentity"
	Using task DetectSigningIdentity from Xamarin.iOS.Tasks.DetectSigningIdentity, Xamarin.iOS.Tasks, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
	The certificate 'Developer ID Application: Xamarin Inc (7V723M9SQ5)' does not match any of the prefixes 'iPhone Developer', 'iOS Development'.
	The certificate 'Developer ID Installer: Xamarin Inc (7V723M9SQ5)' does not match any of the prefixes 'iPhone Developer', 'iOS Development'.
	The certificate 'FARSCAPE' does not match any of the prefixes 'iPhone Developer', 'iOS Development'.
	The certificate 'Mac Developer: Builder Xamarin (C9K3627T5A)' does not match any of the prefixes 'iPhone Developer', 'iOS Development'.
	Available certificates:
	    iPhone Developer: Builder Xamarin (C9K3627T5A)
	    iPhone Developer: Builder Xamarin (RH6G28WM67)
	Available profiles:
	    SJ4M429CQ6_Builder_tvOS_BuildMachine
	'161174814714746EDE30E74E75D9AC9A52F216E1' doesn't match '7D268551AD0F31B49A1FD8588B811D063730B2C2'.
	Finding matching provisioning profiles with compatible AppID, keeping only those with the longest matching (wildcard) IDs.
	Detected signing identity:
	  Code Signing Key: "iPhone Developer: Builder Xamarin (C9K3627T5A)" (161174814714746EDE30E74E75D9AC9A52F216E1)
	  Provisioning Profile: "SJ4M429CQ6_Builder_tvOS_BuildMachine" (e822ee59-a61b-484c-a708-f0de7c938613)
	  Bundle Id: com.xamarin.mytvapp
	  App Id: SJ4M429CQ6.com.xamarin.mytvapp
Done executing task "DetectSigningIdentity"
```

It depends on the set of installed provisioning profiles (which is apparently different between wrench and Jenkins), so it went unnoticed.

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs [hunk 1]
```diff
@@ -304,15 +305,35 @@ protected virtual MobileProvision GetMobileProvision (MobileProvisionPlatform pl
...
+			switch (SdkPlatform) {
+			case "AppleTVSimulator":
+			case "AppleTVOS":
+				platform = MobileProvisionPlatform.tvOS;
+				break;
+			case "iPhoneSimulator":
+			case "WatchSimulator":
+			case "iPhoneOS":
+			case "WatchOS":
+				platform = MobileProvisionPlatform.iOS;
```
Comment (rolfbjarne) [inline msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: Shouldn't you have a `watchOS` case as well?

Comment (jstedfast) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: It already has `case "WatchOS":`

Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: Sorry, I meant to point out that the `WatchOS` case points to `MobileProvisionPlatform.iOS`. Is that intended?

Comment (jstedfast) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: Yes. I don't think there is a watchOS-specific platform value in provisioning profiles.

FWIW, the old code was hard-coded to iOS.

**Comment by rolfbjarne** [comment]
Test failure is unrelated (https://github.com/xamarin/maccore/issues/612)

File: msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs (hunk 1 above)
Comment (spouliot) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: 👍

Comment (jstedfast) [reply msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CompileEntitlementsTaskBase.cs]: It's not clear to me either. Perhaps we just got lucky that the provisioning profiles we had targeted both iOS and tvOS? Or maybe it never worked and we've just been ignoring the warnings this whole time?

**Comment by jstedfast** [comment]
is this safe to merge?

**Comment by spouliot** [comment]
@jstedfast yes, see @rolfbjarne comment above

**Pull request merged**